activerecord/test/cases/reaper_test.rb

```diff
@@ -40,6 +38,8 @@ def test_nil_time
         reaper = ConnectionPool::Reaper.new(fp, nil)
         reaper.run
         assert_not fp.reaped
+      ensure
+        fp.discard!
```

Comment:
With the new class instance variable that stores the pools for reaping, we need to clean up whenever we create a new pool to avoid leaking state in each test case.